.circleci/config.yml

```diff
@@ -104,38 +104,42 @@ jobs:
-            [[ $CIRCLE_NODE_INDEX =~ [012] ]] || RUST_BACKTRACE=1 cargo build -j 16 -p cluster-test
-            [[ $CIRCLE_NODE_INDEX =~ [012] ]] || RUST_BACKTRACE=1 cargo build -j 16 -p libra-fuzzer
-            [[ $CIRCLE_NODE_INDEX =~ [012] ]] || RUST_BACKTRACE=1 cargo build -j 16 -p language_benchmarks
-            [[ $CIRCLE_NODE_INDEX =~ [012] ]] || RUST_BACKTRACE=1 cargo build -j 16 -p bytecode-to-boogie
-            [[ $CIRCLE_NODE_INDEX =~ [012] ]] || RUST_BACKTRACE=1 cargo build -j 16 -p cost-synthesis
-            [[ $CIRCLE_NODE_INDEX =~ [012] ]] || RUST_BACKTRACE=1 cargo build -j 16 -p test-generation
+            [[ $CIRCLE_NODE_INDEX =~ [0124] ]] || RUST_BACKTRACE=1 cargo build -j 16
+            [[ $CIRCLE_NODE_INDEX =~ [0124] ]] || RUST_BACKTRACE=1 cargo build -j 16 -p libra-swarm
+            [[ $CIRCLE_NODE_INDEX =~ [0124] ]] || RUST_BACKTRACE=1 cargo build -j 16 -p cluster-test
+            [[ $CIRCLE_NODE_INDEX =~ [0124] ]] || RUST_BACKTRACE=1 cargo build -j 16 -p libra-fuzzer
+            [[ $CIRCLE_NODE_INDEX =~ [0124] ]] || RUST_BACKTRACE=1 cargo build -j 16 -p language_benchmarks
+            [[ $CIRCLE_NODE_INDEX =~ [0124] ]] || RUST_BACKTRACE=1 cargo build -j 16 -p bytecode-to-boogie
+            [[ $CIRCLE_NODE_INDEX =~ [0124] ]] || RUST_BACKTRACE=1 cargo build -j 16 -p cost-synthesis
+            [[ $CIRCLE_NODE_INDEX =~ [0124] ]] || RUST_BACKTRACE=1 cargo build -j 16 -p test-generation
       - run:
-          name: Run All Unit Tests
+          name: Run All Non Flaky Unit Tests
           command: |
-            [[ $CIRCLE_NODE_INDEX =~ [013] ]] || RUST_BACKTRACE=1 $CI_TIMEOUT cargo x test --unit
+            [[ $CIRCLE_NODE_INDEX =~ [0134] ]] || RUST_BACKTRACE=1 $CI_TIMEOUT cargo test --all-features --workspace --exclude libra-node --exclude libra-crypto --exclude testsuite --exlude consensus
       - run:
           name: Run Cryptography Unit Tests with the formally verified backend
           command: |
-            [[ $CIRCLE_NODE_INDEX =~ [013] ]] || ( RUST_BACKTRACE=1 cd crypto/crypto && $CI_TIMEOUT cargo test --features='std fiat_u64_backend fuzzing' --no-default-features )
+            [[ $CIRCLE_NODE_INDEX =~ [0134] ]] || ( RUST_BACKTRACE=1 cd crypto/crypto && $CI_TIMEOUT cargo test --features='std fiat_u64_backend fuzzing' --no-default-features )
       - run:
           name: Run All End to End Tests
           command: |
-            [[ $CIRCLE_NODE_INDEX =~ [012] ]] || RUST_BACKTRACE=1 $CI_TIMEOUT cargo x test --package testsuite -- --test-threads 1
+            [[ $CIRCLE_NODE_INDEX =~ [0124] ]] || RUST_BACKTRACE=1 $CI_TIMEOUT cargo x test --package testsuite -- --test-threads 1
+      - run:
+          name: Run Quarantined Unit Tests 3 (consensus) times
+          command: |
+            [[ $CIRCLE_NODE_INDEX =~ [0123] ]] || ( RUST_BACKTRACE=1 cargo x test --package consensus ) || ( RUST_BACKTRACE=1 cargo x test --package consensus ) || ( RUST_BACKTRACE=1 cargo x test --package consensus )
```

Comment:
Would it be easier (i know this isn't exactly shell scripting) to be able to use a loop for retries instead of duplicating the commands multiple times with `||`?

Reply:
This is fast and dirty until I get this capability in to x.    Started last night with a "quaratined" flag for packages but don't want to force folks to wait on that.   I could add a loop sure, don't think I need it yet.  I just wanted to get it out the door this morning.

Reply:
If none of the 3 tries succeeds, the job on this CI container will fail.  It will block the land.  I think we decided not to block land on flaky tests ("quarantined").  You can add `|| true` at the end to make it not fail the job.

Reply:
After looking at it the second time, I'd suggest wrapping this in a loop with a overall timeout not exceeding 20 minutes (current e2e workflow time). CI won't send signal to bors until it finishes the workflow, which includes all the jobs. Circle job has a max run time of 5 hrs by default.  This will lengthen the land queue. 

Wrapping in loop would allow us calculate how flaky the `consensus` tests are.  

Unrelated to this PR is we need to extract test metadata (in xml) from cargo test so that we can plug into CircleCI's built-in support for tracking test (https://circleci.com/docs/2.0/collect-test-data/?utm_medium=SEM&utm_source=gnb&utm_campaign=SEM-gnb-DSA-Eng-ni&utm_content=&utm_term=dynamicSearch-&gclid=Cj0KCQjwjcfzBRCHARIsAO-1_Or5KnFUPks9SIiPlPAsQb6skDoNe4mqkH_zB_ty_c38lNhVLwQsubwaAqpNEALw_wcB)